lib/src/models/nodes.ts

```diff
@@ -8,10 +8,10 @@ export interface ContractNode {
   types: TypeNode[];
 }
 
-export interface TypeNode {
+export interface TypeNode<T extends DataType = DataType> {
```

Comment:
what does this do?

Reply:
It means that `TypeNode` can be restricted to a specific subtype:
- `TypeNode` is the same as `TypeNode<DataType>`
- `TypeNode<UnionType>` is specifically containing a `UnionType`